src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +15,34 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(String.format(
+                    "Cannot save country: %s to DB", country.getName()), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return country;
```

Comment:
According to the checklist, you do not need to set an ID for your entity explicitly as Hibernate will handle it. Ensure that the ID generation strategy is correctly configured in the Country entity class.